include/turtle/detail/expectation_template.hpp

```diff
@@ -196,16 +221,41 @@ namespace detail
         {
             return invocation_->verify();
         }
-
+#if defined(MOCK_THREAD_SAFE)
+        bool verify(const boost::shared_ptr< condition_variable> &cv, lock &lk) const
+        {
+            if (timeout_)
+            {
+                while (!invocation_->verify())
```

Comment:
Is this blocking until all the expectations are fulfilled or time out ?
What happens for other objects expectations ? Won't they sometimes succeed when they should have timed out ?